v22.1/security-reference/scram-authentication.md

```diff
@@ -0,0 +1,326 @@
+2. You must block clients from sending plaintext passwords to CockroachDB during authentication by [customizing your cluster's authentication configuration](#configuring-your-cluster-settings-for-strict-credential-isolation) so that only the `scram-sha-256` and `cert-scram-sha-256` methods, and neither `password` nor `cert-password`, are allowed.
+3. After steps (1) and (2) are complete, you must update any previously created passwords with externally generated SCRAM-SHA-256 hashes, and continue to generate your own hashes for all user creation and password update operations.
+
+Note the importance of rotating passwords twice, both in steps (1) and again in (3), with different passwords.  Step (1) ensures that clients at step (2) can continue to connect after non-SCRAM logins are blocked. However, between step (1) and step (2) it is still possible for CockroachDB to receive plaintext passwords from clients, and so the passwords used at step (1) are not fully protected yet. After step (2), CockroachDB will not learn plaintext passwords from clients any more; then after step 3 it will not know plaintext passwords at all.
+
+### Configuring your cluster settings for strict credential isolation
+
+To enable strict credential isolation from CockroachDB, additional cluster settings are required beyond simply enabling SCRAM authentication, as detailed in [Enabling SCRAM-SHA-256 authentication for new users/roles
+](#enabling-scram-sha-256-authentication-for-new-users-roles).
+
+Required cluster settings:
+
+
+#### `server.user_login.password_encryption`
+
+Enable SCRAM for new passwords.
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SET CLUSTER SETTING server.user_login.password_encryption = 'scram-sha-256';
+~~~
+
+#### `server.user_login.store_client_pre_hashed_passwords.enabled`
+
+Enable the CockroachDB cluster to accept pre-computed SCRAM-SCHA-256 salted password hashes, rather than accepting a plaintext password and computing the hash itself.
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SET CLUSTER SETTING setting server.user_login.store_client_pre_hashed_passwords.enabled = true;
+~~~
+
+#### `server.host_based_authentication.configuration`
+
+Allow *only* `scram-sha-256` (not the weaker `password`) in your [Authentication Configuration (HBA)](authentication.html#authentication-configuration).
+
+{{site.data.alerts.callout_danger}}
+After this change, users will not be able to sign in with plaintext passwords. All critical SQL users must be migrated to SCRAM-SHA-256 authentication before making this change:
+1. All client drivers must be updated to a version that supports SCRAM handshakes.
+2. All stored credentials for client apps must be updated to use SCRAM encoded passwords. This procedure is detailed in [Migrating existing users/roles to SCRAM-SHA-256 authentication](#migrating-existing-users-roles-to-scram-sha-256-authentication)
```

Comment:
Missing period at the end of this sentence: "This procedure is detailed in [Migrating existing users/roles to SCRAM-SHA-256 authentication](#migrating-existing-users-roles-to-scram-sha-256-authentication)"